src/tests/ftest/pool/rf.py

```diff
@@ -71,3 +55,115 @@ def test_rf_pool_property(self):
+        cont_prop = re.findall(r"\d+", cont_prop_value)
+        cont_prop_int = int(cont_prop[0])
+        self.assertEqual(cont_prop_int, expected_rd_fac, msg)
+
+    def test_rf_pool_property(self):
+        """Test rd_fac configuration for pool and container.
+
+        Pool's rd_fac is used to define the default rd_fac value of its container. If a
+        container is created without rd_fac during container create, pool's rd_fac will be
+        used. If user wants to set different rd_fac from pool's, they can pass it in as
+        --properties=rd_fac:N during container create.
+
+        Test Description:
+        1. Create a pool with rd_fac passed in.
+        2. Verify pool's rd_fac by calling dmg pool get-prop.
+        3. Verify pool's rd_fac can't be changed to invalid value.
+        4. Verify that pool and container's rd_fac haven't been changed.
+        5. Verify container's default and requested rd_fac.
+        6. Update pool's rd_fac to a valid value and verify.
+        7. Verify container's default and requested rd_fac with updated pool rd_fac.
+        8. Verify that while pool's rd_fac was changed, container's rd_fac remained the
+        same.
+
+        Jira ID: DAOS-9217, DAOS-17768
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=pool,redundancy,redundancy_factor,rf
+        :avocado: tags=PoolRedunFacProperty,test_rf_pool_property
+        """
+        cont_rfs = self.params.get("cont_rf", '/run/container/*')
+
+        # 1. Create a pool with rd_fac passed in.
+        self.log_step("Create a pool with rd_fac passed in.")
+        pool = self.get_pool()
+
+        # 2. Verify pool's rd_fac by calling dmg pool get-prop.
+        self.log_step("Verify pool's rd_fac by calling dmg pool get-prop.")
+        pool_prop_expected = int(pool.properties.value.split(":")[1])
+        self.assertEqual(pool_prop_expected, pool.get_property("rd_fac"))
+
+        # Create a container for later steps.
+        container = self.get_container(pool=pool)
+
+        # 3. Verify pool's rd_fac can't be changed to invalid value.
+        self.log_step("Verify pool's rd_fac can't be changed to invalid value.")
+        invalid_rd_fac = 99
+        dmg_command = self.get_dmg_command()
+        errors = []
+        try:
+            properties = f"rd_fac:{invalid_rd_fac}"
+            dmg_command.pool_set_prop(pool=pool.identifier, properties=properties)
+            self.fail(f"Pool rd_fac was changed to invalid value: {invalid_rd_fac}!")
+        except CommandFailure as command_failure:
+            self.log.info(
+                "Update pool's rd_fac to %s failed as expected", invalid_rd_fac)
+            exp_msg = f"invalid redun fac value {invalid_rd_fac}"
+            if exp_msg not in str(command_failure):
+                msg = (f"Updating pool's rd_fac to invalid value didn't return expected "
+                       f"message! {exp_msg}")
+                errors.append(msg)
```

Comment:
Since this is the only place recording `errors` and everything else fails immediately, do we even need this? Why not just fail?

Reply:
This is just checking whether the expected message was returned, so I thought it's not bad enough to immediately fail. Anyway, I updated to `self.fail()`.